### PR TITLE
Functions/Stability: Convert 'normalize' to FunctionParameter

### PR DESCRIPTION
**Pull request by jvesely**
Convert Stability.normalize to FunctionParameter.

### CODE REVIEW

File: psyneulink/core/components/functions/nonstateful/objectivefunctions.py [hunk 1]
```diff
@@ -205,7 +185,11 @@ class Parameters(ObjectiveFunction.Parameters):
         metric = Parameter(ENERGY, stateful=False)
         metric_fct = Parameter(None, stateful=False, loggable=False)
         transfer_fct = Parameter(None, stateful=False, loggable=False)
-        normalize = Parameter(False, stateful=False)
+        normalize = FunctionParameter(
+            False,
+            function_name='metric_fct',
+            function_parameter_name=NORMALIZE,
```
Comment (kmantel) [inline psyneulink/core/components/functions/nonstateful/objectivefunctions.py:191]: Specifying function_parameter_name isn't needed if it's the same as the parameter name, but it doesn't bother me if this is more readable.

**Review by kmantel — COMMENTED** [review]
I'm not sure if normalize and metric were meant to be there like an interface, where they must also be on any hypothetical future ObjectiveFunctions or not. Or if we intend on making any new ObjectiveFunctions at all. Any thoughts @jdcpni?

But I don't see anything wrong here, so if it simplifies something on the LLVM side, OK by me.

**Comment by jvesely** [comment]
> I'm not sure if normalize and metric were meant to be there like an interface, where they must also be on any hypothetical future ObjectiveFunctions or not. Or if we intend on making any new ObjectiveFunctions at all. Any thoughts @jdcpni?
> 
> But I don't see anything wrong here, so if it simplifies something on the LLVM side, OK by me.

I was just an associated cleanup. The part that matters for LLVM is the conversion of normalize to FunctionParameter, since the parameter is not used in the implementation of Stability.

File: psyneulink/core/components/functions/nonstateful/objectivefunctions.py (hunk 1 above)
Comment (jvesely) [reply psyneulink/core/components/functions/nonstateful/objectivefunctions.py]: removed

**Comment by kmantel** [comment]
> I was just an associated cleanup. I've dropped those parts. The part that matters for LLVM is the conversion of normalize to FunctionParameter, since the parameter is not used in the implementation of Stability.

OK, sounds good.

**Review by kmantel — APPROVED** [review]

**Pull request merged**